scene/gui/code_edit.cpp

```diff
@@ -1154,6 +1154,22 @@ void CodeEdit::_new_line(bool p_split_current_line, bool p_above) {
 			set_caret_line(get_caret_line(i) - 1, false, true, 0, i);
 			set_caret_column(get_line(get_caret_line(i)).length(), i == 0, i);
 		}
+
+		// Insert a documentation comment if the caret was previously on a documentation comment block.
+		int prev_line = get_caret_line(i) + (p_above ? 1 : -1);
+
+		if (is_in_doc_comment(prev_line) != -1) {
+			const String prev_line_content = get_line(prev_line).strip_edges();
+
+			for (const String doc_comment : get_doc_comment_delimiters()) {
+				const String beg = doc_comment.get_slice(" ", 0);
+
+				if (prev_line_content.begins_with(beg)) {
+					insert_text_at_caret(doc_comment + " ", i);
+					break;
+				}
+			}
```

Comment:
There should be a separate kind of key for the delimiters, since other styles have a different characters for start `/**`, end `*/`, and in between new lines `*`.

Maybe the presence of this new line key can be what enables this behavior, and then we may not need a separate doc comment type?